File: dotnet/ArqLibrarianClassic/ArqLibrarianClassic/Library/MemoryUserDao.cs
using System.Collections.Generic;
using System.Linq;

namespace ArqLibrarianClassic.Library
{
    public class MemoryUserDao : UserDao
    {
        private static readonly List<User> users = new List<User>();

        public void Insert(User user)
        {
            user.Id = Generated.UserId();
            users.Add(user);
        }
        
        public User FindById(long id)
        {
            var user = users.FirstOrDefault(u => u.Id == id);

            if (user == null)
            {
                throw new LibrarianException($"User not found with id = {id}");
            }

            return user;
        }

        public void Init()
        {
            Insert(new User("kowalski", "kowal", "Jan Kowalski", "Gdynia", "87052507754"));
            Insert(new User("nowak", "nowypass", "Piotr Nowak", "Warszawa", "890224031121"));
            Insert(new User("koper", "dupadupa", "Wojciech Koperski", "Zakopane", "91121202176"));
        }

        public void Clear()
        {
            users.Clear();
        }
    }
}

File: java/ArqLibraryClassic/ArqLibraryLayered/src/bnsit/ms/arq/library/MemoryUserDao.java
package bnsit.ms.arq.library;

import java.util.ArrayList;
import java.util.List;

public class MemoryUserDao implements UserDao {
    private static final List<User> users = new ArrayList<User>();

    @Override
    public void insert(User user)
    {
        user.setId(Generated.userId());
        users.add(user);
    }

    @Override
    public User findById(long userId) {
        User user = users.stream().filter(u -> u.getId() == userId).findFirst().get();

        if (user == null)
        {
            throw new LibrarianException(String.format("User not found with id = %d", userId));
        }

        return user;
    }

    public void clear() {
        users.clear();
    }

    public void init() {
        insert(new User("kowalski", "kowal", "Jan Kowalski", "Gdynia", "87052507754"));
        insert(new User("nowak", "nowypass", "Piotr Nowak", "Warszawa", "890224031121"));
        insert(new User("koper", "dupadupa", "Wojciech Koperski", "Zakopane", "91121202176"));
    }
}
